python-frontend/typeshed_serializer/tests/test_tox_runner.py

```diff
@@ -0,0 +1,219 @@
+        with mock.patch(self.PATH_IS_FILE_FUNCTION) as mocked_isfile, \
+                mock.patch(self.BUILTIN_OPEN_FUNCTION, mock_open(read_data=file_data)) as mocked_open:
+            mocked_isfile.return_value = True
+            assert tox_runner.read_previous_checksum(checksum_file) == file_data
+            mocked_open.assert_called_with(checksum_file, 'r')
+
+    def test_update_checksum(self):
+        binary_file_names = ['test.protobuf', 'other.protobuf']
+        source_checksum = '123'
+        binaries_checksum = '456'
+        checksums = [source_checksum, binaries_checksum]
+
+        def feed_checksum(_fn, _f):
+            return checksums.pop()
+
+        with mock.patch(self.BUILTIN_OPEN_FUNCTION, mock_open()) as mocked_open, \
+                mock.patch(f'{self.MODULE_NAME}.fetch_binary_file_names') as mock_binary_files, \
+                mock.patch(f'{self.MODULE_NAME}.fetch_source_file_names') as mock_files, \
+                mock.patch(self.COMPUTE_CHECKSUM_FUNCTION) as mocked_checksum:
+            mocked_checksum.side_effect = feed_checksum
+            mock_files.return_value = self.FILE_NAMES
+            mock_binary_files.return_value = binary_file_names
+            tox_runner.update_checksum()
+            mocked_file = mocked_open()
+            mocked_open.assert_any_call(CHECKSUM_FILE, 'w')
+            mocked_checksum.assert_any_call(self.FILE_NAMES, tox_runner.normalize_text_files)
+            mocked_file.write.assert_any_call(source_checksum)
+            mocked_open.assert_any_call(CHECKSUM_BINARIES_FILE, 'w')
+            mocked_checksum.assert_any_call(binary_file_names, tox_runner.read_file)
+            mocked_file.write.assert_any_call(binaries_checksum)
+
+    def test_normalized_text_files_rn(self):
+        with mock.patch(f'{self.MODULE_NAME}.Path.read_text') as mock_read_text:
+            mock_read_text.return_value = "\r\ntest\r\n end\r\n"
+            text = tox_runner.normalize_text_files("test")
+            assert text == self.FILE_CONTENT
+
+    def test_normalized_text_files_r(self):
+        with mock.patch(f'{self.MODULE_NAME}.Path.read_text') as mock_read_text:
+            mock_read_text.return_value = "\rtest\r end\r"
+            text = tox_runner.normalize_text_files("test")
+            assert text == self.FILE_CONTENT
+
+    def test_normalized_text_files(self):
+        with mock.patch(f'{self.MODULE_NAME}.Path.read_text') as mock_read_text:
+            mock_read_text.return_value = "\ntest\n end\n"
+            text = tox_runner.normalize_text_files("test")
+            assert text == self.FILE_CONTENT
+
+    def test_read_file(self):
+        file_bytes = bytes("\ntest end\n", 'utf-8')
+        with mock.patch(f'{self.MODULE_NAME}.Path.read_bytes') as mock_read_bytes:
+            mock_read_bytes.return_value = file_bytes
+            text = tox_runner.read_file("test")
+            assert text == file_bytes
+
+    def test_compute_checksum(self):
```

Comment:
I would like to see a test where we call twice actual files from `tests/resources` to `compute_checksum` and assert the return value is the same between the two calls